Kudu.Services/Arm/ArmControllerConfiguration.cs

```diff
@@ -0,0 +1,14 @@
+using System;
+using System.Web.Http.Controllers;
+
+namespace Kudu.Services.Arm
+{
+    [AttributeUsage(AttributeTargets.Class)]
+    public sealed class ArmControllerConfiguration : Attribute, IControllerConfiguration
+    {
+        public void Initialize(HttpControllerSettings controllerSettings, HttpControllerDescriptor controllerDescriptor)
+        {
+            controllerSettings.Services.Replace(typeof(IHttpActionSelector), new ArmActionSelector());
```

Comment:
Yes, it's tricky. It modifies the name of the action method that we're binding to. So foo --> FooArm. That's why you don't find "arm" literal anywhere else. Definitely calls for comment!